common/djangoapps/util/help_desk.py

```diff
@@ -0,0 +1,443 @@
+            'email': email,
+            'priority': priority,
+            'status': status,
+            'group_id': group_id
+        }
+
+        response = requests.post(
+            settings.HELPDESK_URL + '/api/v2/tickets',
+            auth=(settings.HELPDESK_API_KEY, "unused_but_required"),
+            headers=headers,
+            data=json.dumps(payload),
+            allow_redirects=False)
+
+        if response.ok:
+            message = json.loads(response.content)
+            return message['id']
+        else:
+            raise FreshdeskError(response.content, response.status_code, response)
+
+
+    def update_ticket(self, ticket_id, note, tags):
+        """
+        Update the Freshdeck ticket with id `ticket_id` using the given `note`.
+
+        The update should have the format specified by the freshdesk api.
+        https://freshdesk.com/api#add_note_to_a_ticket
+        """
+        headers = {'Content-Type': 'application/json'}
+        payload = {
+            "body" : note,
+            "private" : True
+        }
+
+        response = requests.post(
+            settings.HELPDESK_URL + '/api/v2/tickets/' + str(ticket_id) + '/notes',
+            auth=(settings.HELPDESK_API_KEY, "unused_but_required"),
+            headers=headers,
+            data=json.dumps(payload),
+            allow_redirects=False)
+
+        if response.ok:
+            return json.loads(response.content)
+        else:
+            raise FreshdeskError(response.content, response.status_code, response)
+
+
+    def get_group(self, name):
+        """
+        Find the Freshdesk group named `name`. Groups are cached for
+        CACHE_TIMEOUT seconds.
+
+        If a matching group exists, it is returned as a dictionary
+        with the format specifed by the freshdesk api.
+
+        Otherwise, returns None.
+        """
+        cache = caches['default']
+        cache_key = '{prefix}_group_{name}'.format(prefix=self.CACHE_PREFIX, name=name)
+        cached = cache.get(cache_key)
+        if cached:
+            return cached
+
+        headers = {'Content-Type': 'application/json'}
+        response = requests.get(
+            settings.HELPDESK_URL + '/api/v2/groups',
+            auth=(settings.HELPDESK_API_KEY, "unused_but_required"),
+            headers=headers,
+            allow_redirects=False)
+
+        groups = json.loads(response.content)
+
+        if response.ok:
+            for group in groups:
+                if group['group']['name'] == name:
+                    cache.set(cache_key, group, self.CACHE_TIMEOUT)
+                    return group
+        else:
+            raise FreshdeskError(groups, response.status_code, response)
+
+        return None
+
+    @abstractmethod
+    def record_feedback(self, *args, **kwargs):
+        """
+        Returns the _record_feedback_in_freshdesk implementation.
+        """
+        return _record_feedback_in_freshdesk(*args, **kwargs)
+
+
+def _record_feedback_in_freshdesk(
```

Comment:
please don't mix single quotes and double quotes, this comment is for the method not line

Reply:
Previous implementation, changed.